File: app/src/main/java/mono/hg/adapters/HiddenAppAdapter.java
package mono.hg.adapters;

import android.content.Context;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

import mono.hg.R;
import mono.hg.models.AppDetail;
import mono.hg.utils.Utils;

public class HiddenAppAdapter extends BaseAdapter {
    private ArrayList<AppDetail> hiddenAppsList;
    private Context context;

    public HiddenAppAdapter(ArrayList<AppDetail> hiddenApps, Context context) {
        this.hiddenAppsList = hiddenApps;
        this.context = context;
    }

    @Override public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        ViewHolder appHolder;
        View view = convertView;

        if (view == null) {
            view = Utils.requireNonNull(inflater).inflate(R.layout.list_generic_item, parent, false);

            appHolder = new ViewHolder();

            appHolder.icon = view.findViewById(R.id.item_icon);
            appHolder.name = view.findViewById(R.id.item_name);

            view.setTag(appHolder);
        } else {
            appHolder = (ViewHolder) view.getTag();
        }

        appHolder.name.setText(hiddenAppsList.get(position).getAppName());

        if (hiddenAppsList.get(position).isAppHidden()) {
            appHolder.icon.setImageResource(R.drawable.ic_check);
            appHolder.name.setTypeface(Typeface.DEFAULT_BOLD);
        } else {
            appHolder.icon.setImageDrawable(hiddenAppsList.get(position).getIcon());
            appHolder.name.setTypeface(Typeface.DEFAULT);
        }

        return view;
    }

    @Override public int getCount() {
        return hiddenAppsList.size();
    }

    @Override public Object getItem(int position) {
        return hiddenAppsList.get(position);
    }

    @Override public long getItemId(int position) {
        return position;
    }

    @Override public boolean hasStableIds() {
        return true;
    }

    private static class ViewHolder {
        private ImageView icon;
        private TextView name;
    }
}

File: app/src/main/java/mono/hg/adapters/HiddenAppAdapter.kt
package mono.hg.adapters

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Typeface
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.core.widget.ImageViewCompat
import mono.hg.R
import mono.hg.helpers.PreferenceHelper
import mono.hg.models.App
import java.util.*

/**
 * Adapter handling display of hidden apps. Only used in preferences.
 */
class HiddenAppAdapter(private val hiddenAppsList: ArrayList<App>, private val context: Context) :
    BaseAdapter() {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        val appHolder: ViewHolder
        var view = convertView

        if (view == null) {
            view = View.inflate(context, R.layout.list_generic_item, null)
            appHolder = ViewHolder()
            appHolder.icon = view.findViewById(R.id.item_icon)
            appHolder.name = view.findViewById(R.id.item_name)
            view.tag = appHolder
        } else {
            appHolder = view.tag as ViewHolder
        }

        appHolder.name?.text = hiddenAppsList[position].appName
        if (hiddenAppsList[position].isAppHidden) {
            appHolder.icon?.setImageResource(R.drawable.ic_check)
            appHolder.icon?.let {
                ImageViewCompat.setImageTintList(
                    it,
                    ColorStateList.valueOf(PreferenceHelper.accent)
                )
            }
            appHolder.name?.typeface = Typeface.DEFAULT_BOLD
        } else {
            appHolder.icon?.setImageDrawable(hiddenAppsList[position].icon)
            appHolder.icon?.let { ImageViewCompat.setImageTintList(it, null) }
            appHolder.name?.typeface = Typeface.DEFAULT
        }
        return view
    }

    override fun getCount(): Int {
        return hiddenAppsList.size
    }

    override fun getItem(position: Int): Any {
        return hiddenAppsList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun hasStableIds(): Boolean {
        return true
    }

    private class ViewHolder {
        internal var icon: ImageView? = null
        internal var name: TextView? = null
    }

}